### ISSUE TITLE
Adding option to remove Dock Icon

### ISSUE THREAD

**Issue by estysdesu** [buttercup/buttercup-desktop#806]
This strictly has to do with **macOS**, I'm not sure how this would affect any other OS. I'm not sure if this is possible from Electron (from my understanding that's what the desktop client is built with), but I believe it is. Is it possible to add a setting to remove the dock icon and and the app window from always being present? Kind of like having the app run in the background with just the menu bar icon. A few apps already do this on mac (with a setting). NordVPN being one, OneDrive (and other cloud syncing apps), etc. 

NordVPN example (the app is running in the background with no open window or dock icon):
![image](https://user-images.githubusercontent.com/37026871/64493304-7859a880-d24c-11e9-87ac-cf6329da25c9.png)

EDIT: This actually already happens if you open Buttercup (have the setting to enable the menubar icon enabled) and then close the window. Clicking the menu bar icon opens the window app again and shows the dock icon which is a good enough amount of functionality I think. It would maybe just be nice to have a setting to set this as a default. I want to set Buttercup to start on startup. But, I don't care for the window popup as I don't always need the app every minute of the day. So just leaving it in the background is ideal.

A setting in this menu would be where it would be implemented I imagine.
![image](https://user-images.githubusercontent.com/37026871/64493356-33824180-d24d-11e9-8b7d-e8c03166a4e6.png)

**Comment by perry-mitchell** [comment]
So you mean a setting to start the app minimised to the menu bar only? I like the idea.. I think it’d be quite an easy addition too.

**Comment by estysdesu** [comment]
@perry-mitchell yes sir. That is what I'm thinking. The underlying functionality seems there, but not on start. Thanks for looking into it.

**Comment by perry-mitchell** [comment]
This is perhaps addressed more concisely in #972, so I'll close in favour of that. This will be added soon.

**Comment by ddrouinqc** [comment]
With the new requirement of needing buttercup to run in the background to use the browser extension, this feature request seems a lot more relevant for user who enjoy an uncluttered dock. Is it still on the radar? (The above linked issue #972 has been deleted)

**Comment by perry-mitchell** [comment]
I'll reopen this. Entirely agree it's a good idea.
